### ISSUE TITLE
Support POCO Command Settings

### ISSUE THREAD

**Issue by owenneil** [spectreconsole/spectre.console#444]
**Is your feature request related to a problem? Please describe.**
I was attempting to create new command settings and thought a record would be a useful way to define the settings. However, records cannot inherit from `CommandSettings`, so this is not currently possible.

**Describe the solution you'd like**
Allow the use of command settings classes without inheriting from `CommandSettings`, either by using an interface like `ICommandSettings` with default interface implementation for validation, or without type restriction and not supporting custom validation.

This would allow me to write a setting class like this:

```csharp
    public record ContextSetSettings(
        [property: 
            CommandArgument(0, "<Name>"),
            Description("Which context name to set in this directory")] string Tenant) // or : ICommandSettings
    {
        [Description("Optional username to use for this context.")]
        [CommandOption("-u|--userhint")]
        public string? UserHint { get; init; }
    }

   public class ContextSetCommand : AsyncCommand<ContextSetSettings>
    {
        public override async Task<int> ExecuteAsync(CommandContext context, ContextSetSettings settings)
        { }
    }
```

**Describe alternatives you've considered**
Use interface with default implementation instead of abstract class.
```csharp
public interface ICommandSettings
{
    ValidationResult Validate() => ValidationResult.Success;
}

public abstract class AsyncCommand<TSettings> : ICommand<TSettings>, ICommandLimiter<TSettings>, ICommand where TSettings : ICommandSettings
```
Or allow complete POCO types without type limitation. 
```csharp
public abstract class AsyncCommand<TSettings> : ICommand<TSettings>, ICommandLimiter<TSettings>, ICommand where TSettings: class
{

}

// in CommandPropertyBinder.cs
// Validate the settings if available.
if (settings is CommandSettings commandSettings)
{
    var validationResult = settings.Validate();
    if (!validationResult.Successful)     
    {      
        throw CommandRuntimeException.ValidationFailed(validationResult); 
     }
}
```

**Comment by phil-scott-78** [comment]
Personally, I'm not in love with that record syntax here but I appreciate where this is coming from.

Unfortunately, I'm not sure the POCO option will work in practice mostly due to how `ICommand` is defined. We'd have to do some significant unraveling of the code to allow it to take any object, I fear. 

So I'm partial to the default interface implementation myself. I think a little rigidity is nice to direct users of the library into getting the command and settings configured properly anyway. The issue we run into is we still target .NetStandard 2.0. I did a quick POC using default interfaces wrapped in an ifdef targeting NET5 and leaving the current implementation for NetStandard 2.0 and it worked ok.  But during the changes I had to make a is a lot of public facing code changes that expects `CommandSettings`, specifically the `LogInterceptor`. I'm fairly sure this will be a breaking change for people writing these, but obviously not a huge one.

But with the changes in place, the unit tests pass and all the CLI demos work fine, and then record syntax is valid.

There's also a top secret third option. We hold onto this idea and come back to it later. We've discussed getting the current build super stable and marked as 1.0, and then dropping NetStandard 2.0 for a new major release. At this point I think using interfaces with a default implementation so that records or classes makes a ton of sense.
